internal/tofu/context_plan.go

```diff
@@ -557,15 +555,30 @@ func (c *Context) findImportTargets(config *configs.Config, priorState *states.S
 	for _, ic := range config.Module.Import {
 		if priorState.ResourceInstance(ic.To) == nil {
 			importTargets = append(importTargets, &ImportTarget{
-				Addr:   ic.To,
-				ID:     ic.ID,
 				Config: ic,
 			})
 		}
 	}
 	return importTargets
 }
 
+func (c *Context) validateImportTargets(config *configs.Config, importTargets []*ImportTarget) (diags tfdiags.Diagnostics) {
+	for _, imp := range importTargets {
+		staticAddress := imp.StaticAddr()
+		descendantConfig := config.Descendent(staticAddress.Module)
+		if descendantConfig == nil {
+			diags = diags.Append(&hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Cannot import to non-existent resource address",
+				Detail:   fmt.Sprintf("Importing to resource address %s is not possible, because that address does not exist in configuration. Please ensure that the resource key is correct, or remove this import block.", staticAddress),
```

Comment:
```suggestion
				Detail:   fmt.Sprintf("Importing to resource address '%s' is not possible, because that address does not exist in configuration. Please ensure that the resource key is correct, or remove this import block.", staticAddress),
```

Reply:
c503f29be56d92f253c22f1cbfcd9b1056177af8